src/pages/iou/request/step/IOURequestStepScan/index.native.js

```diff
@@ -0,0 +1,293 @@
+        // https://github.com/zoontek/react-native-permissions/blob/a836e114ce3a180b2b23916292c79841a267d828/README.md?plain=1#L670
+        CameraPermission.requestCameraPermission()
+            .then((status) => {
+                setCameraPermissionStatus(status);
+
+                if (status === RESULTS.BLOCKED) {
+                    FileUtils.showCameraPermissionsAlert();
+                }
+            })
+            .catch(() => {
+                setCameraPermissionStatus(RESULTS.UNAVAILABLE);
+            });
+    };
+
+    const takePhoto = useCallback(() => {
+        const showCameraAlert = () => {
+            Alert.alert(translate('receipt.cameraErrorTitle'), translate('receipt.cameraErrorMessage'));
+        };
+
+        if (!camera.current) {
+            showCameraAlert();
+            return;
+        }
+
+        camera.current
+            .takePhoto({
+                qualityPrioritization: 'speed',
+                flash: flash ? 'on' : 'off',
+            })
+            .then((photo) => {
+                const filePath = `file://${photo.path}`;
+                IOU.setMoneyRequestReceipt_temporaryForRefactor(transactionID, filePath, photo.path);
+
+                if (backTo) {
+                    Navigation.goBack(backTo);
+                    return;
+                }
+
+                const onSuccess = (receipt) => {
+                    IOU.replaceReceipt(transactionID, receipt, filePath);
+                };
+
+                // When an existing transaction is being edited (eg. not the create transaction flow)
+                if (transactionID !== CONST.IOU.OPTIMISTIC_TRANSACTION_ID) {
+                    FileUtils.readFileAsync(filePath, photo.path, onSuccess);
+                    Navigation.dismissModal();
+                    return;
+                }
+
+                // If a reportID exists in the report object, it's because the user started this flow from using the + button in the composer
+                // inside a report. In this case, the participants can be automatically assigned from the report and the user can skip the participants step and go straight
+                // to the confirm step.
+                if (report.reportID) {
+                    IOU.setMoneyRequestParticipantsFromReport(transactionID, report);
+                    Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(iouType, transactionID, reportID));
+                    return;
+                }
+
+                Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID));
+            })
+            .catch((error) => {
+                showCameraAlert();
+                Log.warn('Error taking photo', error);
+            });
+    }, [flash, iouType, report, translate, transactionID, reportID, backTo]);
+
+    // Wait for camera permission status to render
+    if (cameraPermissionStatus == null) {
+        return null;
+    }
+
+    return (
```

Comment:
This caused a regression in https://github.com/Expensify/App/issues/32831 where the add receipt header wasn't showing, as native doesn't support drag and drop, so we needed to wrap the `View` with `StepScreenWrapper` here.